extensions/src/org/pentaho/platform/plugin/services/importer/SolutionImportHandler.java

```diff
@@ -309,57 +316,100 @@ public void importFile( IPlatformImportBundle bundle ) throws PlatformImportExce
+            // the space(s)
+            if ( jobScheduleRequest.getInputFile().contains( " " ) || jobScheduleRequest.getOutputFile()
+              .contains( " " ) ) {
+              log.info( "Could not import schedule, attempting to replace spaces with underscores and retrying: "
+                + jobScheduleRequest.getInputFile() );
+              File inFile = new File( jobScheduleRequest.getInputFile() );
+              File outFile = new File( jobScheduleRequest.getOutputFile() );
+              String inputFileName = inFile.getParent() + RepositoryFile.SEPARATOR
+                + inFile.getName().replaceAll( " ", "_" );
+              String outputFileName = outFile.getParent() + RepositoryFile.SEPARATOR
+                + outFile.getName().replaceAll( " ", "_" );
+              jobScheduleRequest.setInputFile( inputFileName );
+              jobScheduleRequest.setOutputFile( outputFileName );
+              try {
+                if ( File.separator != RepositoryFile.SEPARATOR ) {
+                  // on windows systems, the backslashes will result in the file not being found in the repository
+                  jobScheduleRequest.setInputFile( inputFileName.replace( File.separator, RepositoryFile.SEPARATOR ) );
+                  jobScheduleRequest
+                    .setOutputFile( outputFileName.replace( File.separator, RepositoryFile.SEPARATOR ) );
+                }
+                Response response = createSchedulerJob( schedulerResource, jobScheduleRequest );
+                if ( response.getStatus() == Response.Status.OK.getStatusCode() ) {
+                  if ( response.getEntity() != null ) {
+                    // get the schedule job id from the response and add it to the import session
+                    ImportSession.getSession().addImportedScheduleJobId( response.getEntity().toString() );
+                  }
                 }
+              } catch ( Exception ex ) {
+                // log it and keep going. we should stop processing all schedules just because one fails.
+                log.error( Messages.getInstance()
+                  .getString( "SolutionImportHandler.ERROR_0001_ERROR_CREATING_SCHEDULE", e.getMessage() ), ex );
               }
-            } catch ( Exception ex ) {
+            } else {
               // log it and keep going. we should stop processing all schedules just because one fails.
               log.error( Messages.getInstance()
-                  .getString( "SolutionImportHandler.ERROR_0001_ERROR_CREATING_SCHEDULE", e.getMessage() ), ex );
-            }
-          } else {
-            // log it and keep going. we should stop processing all schedules just because one fails.
-            log.error( Messages.getInstance()
                 .getString( "SolutionImportHandler.ERROR_0001_ERROR_CREATING_SCHEDULE", e.getMessage() ) );
+            }
           }
+        } else {
+          log.info( "Schedule already exist" );
```

Comment:
Please don't add new hard-coded messages. a few lines above you can see that we're already using a message bundle here. Please make sure this file has *no* hard-coded message strings.